src/typing/type_inference_js.ml

```diff
@@ -970,7 +965,7 @@ and statement_decl cx = Ast.Statement.(
       let _, { Ast.Identifier.name; _ } = id in
       let r = mk_reason (spf "class %s" name) loc in
       let tvar = Flow_js.mk_tvar cx r in
-      Env_js.init_env cx name (create_env_entry tvar tvar (Some loc))
+      Env_js.init_env cx name (create_env_entry tvar tvar (Some loc)) VarScope (* TODO: hoist behavior *)
```

Comment:
I'd agree and say that interface definitions should be hoisted for the same reason type aliases should (mutual recursion is important there).

History lesson: The reason classes aren't hoisted is because of hazards with extends clauses. For example, this would be a problem if class decls were hoisted:

```
var _baseClass = SomeBaseClass;
class Foo extends _baseClass {}
```